src/modules/gimbal/output.h

```diff
@@ -55,7 +55,7 @@ class OutputBase
 	explicit OutputBase(const Parameters &parameters);
 	virtual ~OutputBase() = default;
 
-	virtual void update(const ControlData &control_data, bool new_setpoints) = 0;
+	virtual void update(ControlData &control_data, bool new_setpoints) = 0;
```

Comment:
I'd rather keep this const and add a separate getter for the id. It makes it clearer which state the output can change.

Reply:
Right ok. Where would you put the ID and how would you pass it in?

Reply:
The placement in ControlData is fine, but you could also add it as separate ref arg to the method.